johnzon-mapper/src/main/java/org/apache/johnzon/mapper/Mapper.java

```diff
@@ -138,9 +140,9 @@ public void writeObject(final Object object, final Writer stream) {
     private boolean isDeduplicateObjects(Class<?> rootType) {
         Boolean dedup = config.isDeduplicateObjects();
         if (dedup == null) {
-            Mappings.ClassMapping classMapping = mappings.findOrCreateClassMapping(rootType);
-            if (classMapping != null) {
-                dedup = classMapping.isDeduplicateObjects();
+            Mappings.TypeMapping classMapping = mappings.findOrCreateTypeMapping(rootType);
+            if (classMapping != null && MappingType.CLASSMAPPING.equals(classMapping.getType())) {
```

Comment:
just a style thing but maybe add in TypeMapping a as(Class<T>) to be able to do

    typeMapping.as(ClassMapping.class)

it avoids to put cast everywhere and makes the pattern if (type == xxx) tm.as(X.class) more natural IMHO

Reply:
done